src/components/dialog-parameters/parameters.js

```diff
@@ -0,0 +1,220 @@
+import { useSnackbar } from 'notistack';
+
+import { updateConfigParameter } from '../../utils/rest-api';
+import {
+    displayErrorMessageWithSnackbar,
+    useIntlRef,
+} from '../../utils/messages';
+import { SingleLineDiagramParameters } from './single-line-diagram-parameter';
+import { MapParameters } from './map-parameter';
+import { LoadFlow } from './load-flow-parameter';
+import { NetworkParameters } from './network-parameter';
+
+export const useStyles = makeStyles((theme) => ({
+    title: {
+        padding: theme.spacing(2),
+    },
+    grid: {
+        padding: theme.spacing(2),
+    },
+    minWidthMedium: {
+        minWidth: theme.spacing(20),
+    },
+    controlItem: {
+        justifyContent: 'flex-end',
+    },
+    button: {
+        marginBottom: '30px',
+    },
+    advancedParameterButton: {
+        marginTop: theme.spacing(3),
+        marginBottom: theme.spacing(1),
+    },
+}));
+
+export function useParameterState(paramName) {
+    const intlRef = useIntlRef();
+
+    const { enqueueSnackbar } = useSnackbar();
+
+    const paramGlobalState = useSelector((state) => state[paramName]);
+
+    const [paramLocalState, setParamLocalState] = useState(paramGlobalState);
+
+    useEffect(() => {
+        setParamLocalState(paramGlobalState);
+    }, [paramGlobalState]);
+
+    const handleChangeParamLocalState = useCallback(
+        (value) => {
+            setParamLocalState(value);
+            updateConfigParameter(paramName, value).catch((errorMessage) => {
+                setParamLocalState(paramGlobalState);
+                displayErrorMessageWithSnackbar({
+                    errorMessage: errorMessage,
+                    enqueueSnackbar: enqueueSnackbar,
+                    headerMessage: {
+                        headerMessageId: 'paramsChangingError',
+                        intlRef: intlRef,
+                    },
+                });
+            });
+        },
+        [
+            paramName,
+            enqueueSnackbar,
+            intlRef,
+            setParamLocalState,
+            paramGlobalState,
+        ]
+    );
+
+    return [paramLocalState, handleChangeParamLocalState];
+}
+
+export function MakeSwitch(prop, label, callback) {
+    const classes = useStyles();
+    return (
+        <>
+            <Grid item xs={8}>
+                <Typography component="span" variant="body1">
+                    <Box fontWeight="fontWeightBold" m={1}>
+                        <FormattedMessage id={label} />
+                    </Box>
+                </Typography>
+            </Grid>
+            <Grid item container xs={4} className={classes.controlItem}>
+                <Switch
+                    checked={prop}
+                    onChange={callback}
+                    value={prop}
+                    inputProps={{ 'aria-label': 'primary checkbox' }}
+                />
+            </Grid>
+        </>
+    );
+}
+
+export function MakeButton(callback, label) {
```

Comment:
would be better like this (and more react like)
const LabelledButton = ({callback, label}) => {

Reply:
it's fixed